File: src/test/java/nl/rabobank/kotlinmovement/recipes/testutil/TestData.java
package nl.rabobank.kotlinmovement.recipes.testutil;

import nl.rabobank.kotlinmovement.recipes.domain.IngredientRequestJ;
import nl.rabobank.kotlinmovement.recipes.domain.IngredientTypeJ;
import nl.rabobank.kotlinmovement.recipes.domain.RecipeRequestJ;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import java.util.Set;

public class TestData {

    public final static RecipeRequestJ pizzaRecipeJ =  new RecipeRequestJ("Pizza", getDefaultIngredientRequests());

    public static Set<IngredientRequestJ> getDefaultIngredientRequests() {
        return Set.of(
                new IngredientRequestJ("Flower", IngredientTypeJ.DRY, 1000),
                new IngredientRequestJ("Water", IngredientTypeJ.WET, 8000),
                new IngredientRequestJ("Salt", IngredientTypeJ.DRY, 20),
                new IngredientRequestJ("Yeast", IngredientTypeJ.DRY, 2)

        );
    }

    public static RecipeRequestJ peperoniPizzaRecipeJ() {
        final Set<IngredientRequestJ> ingredients = Set.of(
                new IngredientRequestJ("Flower", IngredientTypeJ.DRY, 1000),
                new IngredientRequestJ("Water", IngredientTypeJ.WET, 8000),
                new IngredientRequestJ("Salt", IngredientTypeJ.DRY, 20),
                new IngredientRequestJ("Yeast", IngredientTypeJ.DRY, 2),
                new IngredientRequestJ("Peperoni", IngredientTypeJ.DRY, 100),
                new IngredientRequestJ("Tomato sauce", IngredientTypeJ.WET, 100)

        );
        final String newRecipeName = "Pizza Peperoni";
        return new RecipeRequestJ(newRecipeName, ingredients);
    }
    public static List<RecipeRequestJ> generateRecipeRequest(int i) {
        final var recipeRequests = new ArrayList<RecipeRequestJ>(1);
        final String generatedString = generateRandomString();
        while (i > 0) {
            i--;
            recipeRequests.add(new RecipeRequestJ(generatedString, getDefaultIngredientRequests()));
        }
        return recipeRequests;
    }

    public static String generateRandomString() {
        final var random = new Random();
        return random.ints(97, 122 + 1)
                .limit(10)
                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
                .toString();
    }

}

File: src/test/kotlin/nl/rabobank/kotlinmovement/recipes/testutil/testdata/TestData.kt
package nl.rabobank.kotlinmovement.recipes.testutil.testdata

import java.util.*

val defaultIngredientRequests: Set<IngredientRequest> = mutableSetOf(
    IngredientRequest("Flower", IngredientType.DRY, 1000),
    IngredientRequest("Water", IngredientType.WET, 8000),
    IngredientRequest("Salt", IngredientType.DRY, 20),
    IngredientRequest("Yeast", IngredientType.DRY, 2)
)

val pizzaRecipe: RecipeRequest = RecipeRequest("Pizza", defaultIngredientRequests)


fun peperoniPizzaRecipeRequest(): RecipeRequest = mutableSetOf(
    IngredientRequest("Flower", IngredientType.DRY, 1000),
    IngredientRequest("Water", IngredientType.WET, 8000),
    IngredientRequest("Salt", IngredientType.DRY, 20),
    IngredientRequest("Yeast", IngredientType.DRY, 2),
    IngredientRequest("Peperoni", IngredientType.DRY, 100),
    IngredientRequest("Tomato sauce", IngredientType.WET, 100)
).let {
    RecipeRequest("Pizza Peperoni", it)
}

fun generateRecipeRequests(i: Int): List<RecipeRequest> = (0..i).map {
    RecipeRequest(generateRandomString(), defaultIngredientRequests)
}

private fun generateRandomString(): String {
    val random = Random()
    return random.ints(97, 122 + 1)
        .limit(10)
        .collect(
            { StringBuilder() },
            { obj: StringBuilder, codePoint: Int -> obj.appendCodePoint(codePoint) }
        ) { obj: StringBuilder, s: StringBuilder? -> obj.append(s) }
        .toString()
}

